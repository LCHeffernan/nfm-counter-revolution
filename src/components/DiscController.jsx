import React, { useState } from "react";
import PropTypes from "prop-types";
import Disc from "./Disc";

const DiscController = ({ incrementIndex, figures, index }) => {
  const [isButtonDisabled, setIsButtonDisabled] = useState(true);

  const handleClick = () => {
    incrementIndex();
    setIsButtonDisabled(true);
  };
  return (
    <div className="revolver" data-testid="revolver">
      <Disc
        figures={figures}
        setIsButtonDisabled={setIsButtonDisabled}
        index={index}
      />
      <button disabled={isButtonDisabled} onClick={handleClick} type="button">
        increment index
      </button>
    </div>
  );
};

DiscController.propTypes = {
  index: PropTypes.number.isRequired,
  figures: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      title: PropTypes.string,
      angle: PropTypes.number,
      text: PropTypes.arrayOf(PropTypes.string),
    })
  ).isRequired,
  incrementIndex: PropTypes.func.isRequired,
};

export default DiscController;
