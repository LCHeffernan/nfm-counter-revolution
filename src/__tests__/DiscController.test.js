import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import DiscController from "../components/DiscController";

const validProps = {
  index: 0,
  incrementIndex: jest.fn(),
  figures: [
    {
      id: 0,
      title: "figure 0",
      angle: 0,
      text: [
        "Figure 0 Paragraph 1",
        "Figure 0 Paragraph 2",
        "Figure 0 Paragraph 3",
      ],
    },
    {
      id: 1,
      title: "figure 1",
      angle: -60,
      text: [
        "Figure 1 Paragraph 1",
        "Figure 1 Paragraph 2",
        "Figure 1 Paragraph 3",
      ],
    },
    {
      id: 2,
      title: "figure 2",
      angle: -120,
      text: [
        "Figure 2 Paragraph 1",
        "Figure 2 Paragraph 2",
        "Figure 2 Paragraph 3",
      ],
    },
    {
      id: 3,
      title: "figure 3",
      angle: -180,
      text: [
        "Figure 3 Paragraph 1",
        "Figure 3 Paragraph 2",
        "Figure 3 Paragraph 3",
      ],
    },
    {
      id: 4,
      title: "figure 4",
      angle: -240,
      text: [
        "Figure 4 Paragraph 1",
        "Figure 4 Paragraph 2",
        "Figure 4 Paragraph 3",
      ],
    },
    {
      id: 5,
      title: "figure 5",
      angle: -300,
      text: [
        "Figure 5 Paragraph 1",
        "Figure 5 Paragraph 2",
        "Figure 5 Paragraph 3",
      ],
    },
  ],
};

const renderDiscController = () => render(<DiscController {...validProps} />);

describe("DiscController", () => {
  it("matches snapshot", () => {
    const { asFragment } = renderDiscController();

    expect(asFragment()).toMatchSnapshot();
  });

  beforeEach(() => renderDiscController());

  it("renders a Disc component", () => {
    expect(screen.getByTestId("disc")).toBeInTheDocument();
  });

  it("renders a button", () => {
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  it("clicking button calls incrementIndex function", () => {
    const button = screen.getByRole("button");

    fireEvent.click(button);

    expect(validProps.incrementIndex).toHaveBeenCalled();
  });
});
