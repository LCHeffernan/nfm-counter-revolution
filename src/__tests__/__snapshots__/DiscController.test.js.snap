// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscController matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="disc-controller"
    data-testid="revolver"
  >
    <div
      class="disc"
      data-testid="disc"
    >
      <img
        alt=""
        class="disc__layer disc__artwork"
        src="art.png"
      />
      <img
        alt=""
        class="disc__layer disc__mask mask-fade-in"
      />
      <img
        alt=""
        class="disc__layer disc__mask mask-fade-out"
      />
      <img
        alt=""
        class="disc__layer disc__mask mask-fade-out"
      />
      <img
        alt=""
        class="disc__layer disc__mask mask-fade-out"
      />
      <img
        alt=""
        class="disc__layer disc__mask mask-fade-out"
      />
      <img
        alt=""
        class="disc__layer disc__mask mask-fade-out"
      />
    </div>
    <button
      type="button"
    >
      increment index
    </button>
  </div>
</DocumentFragment>
`;
